passes/pmgen/peepopt_muldiv_c.pmg

```diff
@@ -0,0 +1,76 @@
+pattern muldiv_c
+//
+// Authored by Akash Levy of Silimate, Inc. under ISC license.
+// Transforms mul->div into const->mul when b and c are divisible constants:
+// y = (a * b_const) / c_const   ===>   a * eval(b_const / c_const)
+//
+
+state <SigSpec> a b_const mul_y
+
+match mul
+	// Select multiplier
+	select mul->type == $mul
+endmatch
+
+code a b_const mul_y
+	// Get multiplier signals
+	a = port(mul, \A);
+	b_const = port(mul, \B);
+	mul_y = port(mul, \Y);
+
+	// Fanout of each multiplier Y bit should be 1 (no bit-split)
+	for (auto bit : mul_y)
+		if (nusers(bit) != 2)
+			reject;
+
+	// A and B can be interchanged
+	branch;
+	std::swap(a, b_const);
+endcode
+
+match div
```

Comment:
No check that the divider and multiplier are connected

Reply:
I just pulled in https://github.com/akashlevy/yosys/pull/2 which should resolve this. Let me know if this looks good